### ISSUE TITLE
/data decrypt support?

### ISSUE THREAD

**Issue by strlcat** [z3ntu/android_device_fairphone_FP4#2]
Hi,

Built and booted image successfully on my brand new unlocked FP4, which already has some user settings and apps inside. But there is no option to enter password to gain access to /data/data etc.
Is it actually hard to implement?
I don't know much about TWRP internals regarding this.
Previously I owned Pixel 4a, built TWRP for it and it had this option and I could access encrypted files.

Thanks.

**Comment by z3ntu** [comment]
Hi, there's a guide at https://github.com/TeamWin/android_device_qcom_twrp-common , I've attempted it in the past (and to my knowledge also some other people for FP4) but we couldn't get it working so far.

**Comment by strlcat** [comment]
Oh well. I'll look into this closer and will try to compare with pixel 4a tree. Thanks

**Comment by strlcat** [comment]
So well I tried it but I believe I messed with .mk files somewhere, because resulting recovery.img produces log like
```
Using drm graphics.
I:TWFunc::Set_Brightness: Setting brightness control to 127
I:Loading package: splash (/twres/splash.xml)
I:Load XML directly
I:PageManager::LoadFileToBuffer loading filename: '/twres/splash.xml' directly
E:Package splash failed to load.
E:Failed to load splash screen XML.
I:Loading package: TWRP (/TWRP/theme/ui.zip)
I:Loading zip theme
I:Loading package: TWRP (/twres/ui.xml)
I:Load XML directly
E:LoadLanguageListDir '/twres/languages/' path not found
I:PageManager::LoadFileToBuffer loading filename: '/twres/languages/en.xml' directly
I:PageManager::LoadFileToBuffer loading filename: '/twres/ui.xml' directly
E:Package TWRP failed to load.
Failed to load base packages.
E:An internal error has occurred: unable to load theme.
```
endlessly and `recovery` process is crashing with SEGV. I also captured it's strace confirming the issue. By some reason `/twres` directory is whole omitted when I try to build with configuration example from link. If I revert and delete `out` and rebuild, `/twres` returns, TWRP boots fine.

I will look closer tomorrow.

**Comment by z3ntu** [comment]
I believe logcat also should exist in the build, check that also.

**Comment by strlcat** [comment]
Nah, just build system fault. Complete wipe and start from scratch resolved problem. But crypto related services still wont show up, no info even in logcat. Suspecting missing props. Need to compare to reference trees.

**Comment by ghost** [comment]
Do we have any progress here with encrypted user data?

**Comment by z3ntu** [comment]
Not that I know of.

**Comment by VirusBurst** [comment]
https://github.com/eudaldgr/android_device_fairphone_FP4/tree/twrp-11

Just found this source but unfortunately, I am not so far into android cooking to compile TWRP on my own... 
It looks like, someone made an attempt for decryption but... 

Could you maybe compile a TWRP image and post it for testing purposes? I'd like to try
